.github/workflows/.templates/build_packages.yaml

```diff
@@ -0,0 +1,502 @@
+      with:
+        name: source
+        path: .
+    - name: unzip source code
+      run: unzip -q source.zip
+    - name: downloads emqx zip packages
+      env:
+        PROFILE: ${{ matrix.profile }}
+        ARCH: ${{ matrix.arch }}
+        SYSTEM: ${{ matrix.os }}
+      run: |
+        set -e -u -x
+        cd source
+        if [ $PROFILE = "emqx" ];then broker="emqx-ce"; else broker="$PROFILE"; fi
+        if [ $PROFILE = "emqx-ee" ];then edition='enterprise'; else edition='opensource'; fi
+
+        vsn="$(./pkg-vsn.sh)"
+        pre_vsn="$(echo $vsn | grep -oE '^[0-9]+.[0-9]')"
+        if [ $PROFILE = "emqx-ee" ]; then
+            old_vsns=($(git tag -l "e$pre_vsn.[0-9]" | sed "s/e$vsn//"))
+        else
+            old_vsns=($(git tag -l "v$pre_vsn.[0-9]" | sed "s/v$vsn//"))
+        fi
+
+        mkdir -p _upgrade_base
+        cd _upgrade_base
+        for tag in ${old_vsns[@]};do
+          if [ ! -z "$(echo  $(curl -I -m 10 -o /dev/null -s -w %{http_code} https://s3-${{ secrets.AWS_DEFAULT_REGION }}.amazonaws.com/${{ secrets.AWS_S3_BUCKET }}/$broker/$tag/$PROFILE-$SYSTEM-${tag#[e|v]}-$ARCH.zip) | grep -oE "^[23]+")" ];then
+            wget --no-verbose https://s3-us-west-2.amazonaws.com/packages.emqx/$broker/$tag/$PROFILE-$SYSTEM-${tag#[e|v]}-$ARCH.zip
+            wget --no-verbose https://s3-us-west-2.amazonaws.com/packages.emqx/$broker/$tag/$PROFILE-$SYSTEM-${tag#[e|v]}-$ARCH.zip.sha256
+            echo "$(cat $PROFILE-$SYSTEM-${tag#[e|v]}-$ARCH.zip.sha256) $PROFILE-$SYSTEM-${tag#[e|v]}-$ARCH.zip" | sha256sum -c || exit 1
+          fi
+        done
+        cd -
+    - name: build emqx packages
+      env:
+        ERL_OTP: erl23.2.7.2-emqx-2
+        PROFILE: ${{ matrix.profile }}
+        ARCH: ${{ matrix.arch }}
+        SYSTEM: ${{ matrix.os }}
+      run: |
+        set -e -u
+        cd source
+        docker buildx build --no-cache \
+          --platform=linux/$ARCH \
+          -t cross_build_emqx_for_$SYSTEM \
+          -f .ci/build_packages/Dockerfile \
+          --build-arg BUILD_FROM=emqx/build-env:$ERL_OTP-$SYSTEM \
+          --build-arg EMQX_NAME=$PROFILE \
+          --output type=tar,dest=/tmp/cross-build-$PROFILE-for-$SYSTEM.tar .
+
+        mkdir -p /tmp/packages/$PROFILE
+        tar -xvf /tmp/cross-build-$PROFILE-for-$SYSTEM.tar --wildcards emqx/_packages/$PROFILE/*
+        mv emqx/_packages/$PROFILE/* /tmp/packages/$PROFILE/
+        rm -rf /tmp/cross-build-$PROFILE-for-$SYSTEM.tar
+
+        docker rm -f $(docker ps -a -q)
+        docker volume prune -f
+    - name: create sha256
+      env:
+        PROFILE: ${{ matrix.profile}}
+      run: |
+        if [ -d /tmp/packages/$PROFILE ]; then
+          cd /tmp/packages/$PROFILE
+            for var in $(ls emqx-* ); do
+              bash -c "echo $(sha256sum $var | awk '{print $1}') > $var.sha256"
+            done
+          cd -
+        fi
+    - uses: actions/upload-artifact@v1
+      if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Every time you to upload slim build artifact, because testers need to download the latest version